### PR TITLE
Fixed holding retry hotkey at end of map causing hard crash

### LINKED ISSUE #1943: Unexpected behavior when restarting with tilde at end of beatmap
Usually Osu will crash; sometimes the results screen will display behind the now-restarted beatmap.
Steps:
 - Start a map
 - Hold tilde down as the map is ending
 - If timed correctly, the game will behave as described above.

May be related to #1020 , although I could not reproduce that bug.

### PR DESCRIPTION
**Pull request by ThePyrotechnic**
fixes #1943

### CODE REVIEW

**Comment by peppy** [comment]
Can you not just check if `IsCurrentScreen`?

**Comment by ThePyrotechnic** [comment]
Thanks, don't know how I missed that while reading through Screen. Using `IsCurrentScreen` will fix the crash, but still need the other bool for blocking the incorrect rendering:
screenshot 12

**Comment by peppy** [comment]
I'm sure we can figure this out without adding more `bool`s. Can you explain how you get that above scenario to happen?

**Comment by ThePyrotechnic** [comment]
Yep, to get that screen specifically:
 - add the `IsCurrentScreen` check to avoid getting the crash.
 - Load up a short map (I have been able to reproduce this on any map I've tried)
 - AFTER hitting the last note, as the screen begins to fade, hold tilde
 - If the timing is right, the map will restart and the results screen will display.

**Comment by ThePyrotechnic** [comment]
Looks like `AllowPause` will work. I thought I had tested that earlier, but I guess not!

**Comment by peppy** [comment]
Do note that `AllowPause` is not a dynamic variable, and only used in visual tests.

**Comment by ThePyrotechnic** [comment]
I see. In any case I just got it to reoccur with `AllowPause`, so I'll have to keep looking for an alternative.

**Comment by peppy** [comment]
Did you try just using `IsCurrentScreen` in both places?

**Comment by ThePyrotechnic** [comment]
Looks good! Seems to be working fine now.

File: osu.Game/Screens/Play/Player.cs [hunk 1]
```diff
@@ -208,10 +208,12 @@ private void load(AudioManager audio, OsuConfigManager config, APIAccess api)
                 new HotkeyRetryOverlay
                 {
                     Action = () => {
-                        //we want to hide the hitrenderer immediately (looks better).
-                        //we may be able to remove this once the mouse cursor trail is improved.
-                        RulesetContainer?.Hide();
-                        Restart();
+                        if (IsCurrentScreen) {
```
Comment (peppy) [inline osu.Game/Screens/Play/Player.cs]: could you please invert both of these conditionals (ie. `if (!IsCurrentScreen) return;`)?

**Review by peppy — APPROVED** [review]

**Pull request merged**